packages/zowe-explorer/src/zosconsole/ZosConsolePanel.ts

```diff
@@ -0,0 +1,140 @@
+import { randomUUID } from "crypto";
+import Mustache = require("mustache");
+
+export class ZosConsoleViewProvider implements vscode.WebviewViewProvider {
+    public static readonly viewType = "zosconsole";
+
+    private _view?: vscode.WebviewView;
+
+    private profiles: Map<string, imperative.IProfileLoaded> = new Map();
+    private defaultProfileName: string | undefined;
+
+    public constructor(private _extensionUri: vscode.Uri) {
+        this.refreshProfileList();
+    }
+
+    private refreshProfileList(): void {
+        const profileNamesList = ProfileManagement.getRegisteredProfileNameList(globals.Trees.MVS);
+        this.defaultProfileName = profileNamesList[0];
+
+        const loadedProfiles = Profiles.getInstance().allProfiles;
+        loadedProfiles.forEach((profile) => {
+            if (profileNamesList.includes(profile.name)) {
+                this.profiles.set(profile.name, profile);
+            }
+        });
+    }
+
+    public resolveWebviewView(
+        webviewView: vscode.WebviewView,
+        _context: vscode.WebviewViewResolveContext<unknown>,
+        _token: vscode.CancellationToken
+    ): void | Thenable<void> {
+        this._view = webviewView;
+
+        webviewView.webview.options = {
+            enableScripts: true,
+            localResourceRoots: [this._extensionUri],
+        };
+
+        webviewView.webview.html = this._getHtmlForWebview(webviewView.webview);
+
+        webviewView.webview.onDidReceiveMessage(async (message: any) => {
+            const command = message.command;
+            const text = message.text;
+            const profile = message.profile;
+
+            switch (command) {
+                case "startup": {
+                    this.refreshProfileList();
+                    const profileArray = [];
+                    for (const profileName of this.profiles.keys()) {
+                        profileArray.push(profileName);
+                    }
+                    webviewView.webview.postMessage({
+                        type: "optionsList",
+                        profiles: profileArray,
+                        defaultProfile: this.defaultProfileName,
+                    });
+                    break;
+                }
+                case "opercmd":
+                    webviewView.webview.postMessage({
+                        type: "commandResult",
+                        cmd: text,
+                        profile: profile,
+                        result: await this.runOperCmd(text, profile),
+                    });
+                    return;
+            }
+        });
+    }
+
+    private _getHtmlForWebview(webview: vscode.Webview): string {
+        const scriptUri = webview.asWebviewUri(vscode.Uri.joinPath(this._extensionUri, "src", "webviews", "dist", "zos-console", "zos-console.js"));
+        const codiconsUri = webview.asWebviewUri(vscode.Uri.joinPath(this._extensionUri, "node_modules", "@vscode/codicons", "dist", "codicon.css"));
```

Comment:
For icons to be found in VSIX, we need to use the location where Vite is bundling the codicons assets 😁 
```suggestion
        const codiconsUri = webview.asWebviewUri(vscode.Uri.joinPath(this._extensionUri, "src", "webviews", "dist", "codicons", "codicon.css"));
```